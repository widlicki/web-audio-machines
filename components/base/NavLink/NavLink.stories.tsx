import { ComponentMeta, ComponentStory } from '@storybook/react'
import NavLink from './NavLink'

export default {
  title: 'components/base/NavLink',
  component: NavLink,
} as ComponentMeta<typeof NavLink>

const Template: ComponentStory<typeof NavLink> = (args) => (
  <NavLink {...args}>Link</NavLink>
)

export const DefaultNavLink = Template.bind({})

// export const DisabledButton = Template.bind({})
// DisabledButton.args = {
//   disabled: true,
// }
