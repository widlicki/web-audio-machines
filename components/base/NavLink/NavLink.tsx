import { Link } from '@mui/material'
import { useRouter } from 'next/router'
import theme from '../../../styles/Theme'

export interface NavLinkProps {
  to: string
  children?: JSX.Element | string
  props?: any
}

const NavLink = ({ to, children, ...props }: NavLinkProps) => {
  const router = useRouter()

  const handleClick = (e) => {
    e.preventDefault()
    router.push(to)
  }

  return (
    <Link
      sx={{
        fontSize: 16,
        color: 'common.white',
        marginLeft: theme.spacing(3),
        textDecoration: 'none',
        '&:hover': {
          color: theme.palette.primary.dark,
          cursor: 'pointer',
        },
      }}
      {...props}
      onClick={handleClick}
    >
      {children}
    </Link>
  )
}

export default NavLink
