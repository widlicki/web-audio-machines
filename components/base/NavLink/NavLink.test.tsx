import { render } from '@testing-library/react'
import NavLink from './NavLink'
import { DefaultNavLink } from './NavLink.stories'

describe('NavLink Stories', () => {
  new Map([
    ['default NavLink', DefaultNavLink],
    // ['disabled button', DisabledButton],
  ]).forEach((key) => {
    it(`renders ${key}`, () => {
      const component = <NavLink to="/">Link</NavLink>
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})
