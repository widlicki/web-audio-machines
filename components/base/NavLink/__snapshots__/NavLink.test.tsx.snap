// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink Stories renders function () { [native code] } 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-kq33hk-MuiTypography-root-MuiLink-root"
  >
    Link
  </a>
</div>
`;
